### PR TITLE
Fix#66 return by seed

### LINKED ISSUE #54: Return test error for tabular benchmarks/nasbench101
Both benchmark families, nasbench_101 and tabular_benchmarks,  should return test error for objective_function_test instead of the validation error on the highest budget.

### LINKED ISSUE #66: Return performance of random trial for NAS benchmarks
Our current setup by default returns the average across trials for Tabular Benchmarks if more than one trial is available. A more realistic setup would be to also offer the possibility to return the results of a random trial, e.g. if the run_index/data_seed is set to None.

### LINKED ISSUE #67: Return the sum of all trials for tabular_benchmarks
Our objective_function_**test()** for FCNetBaseBenchmark currently by default returns the average performance across trials (using external code). To have a more realistic behaviour, this should be the sum of all trainingcosts.

(the same applies for NB101)

### PR DESCRIPTION
**Pull request by PhMueller**
This pull request fixes several problems with the NAS benchmarks.

### CODE REVIEW

**Comment by PhMueller** [comment]
I am currently uploading the new container for the tabular benchmark. After it is pushed, I will start the tests for this PR.

File: hpobench/benchmarks/nas/nasbench_101.py [hunk 1]
```diff
@@ -109,17 +132,52 @@ def objective_function(self, configuration: Union[CS.Configuration, Dict],
...
+        if isinstance(run_index, int):
+            assert 0 <= run_index <= 2, f'run_index must be in [0, 2], not {run_index}'
+            run_index = (run_index, )
+        elif isinstance(run_index, tuple) or isinstance(run_index, list):
+            assert len(run_index) != 0, 'run_index must not be empty'
+            assert min(run_index) >= 0 and max(run_index) <= 2, \
+                f'all run_index values must be in [0, 2], but were {run_index}'
+        elif run_index is None:
+            logger.debug('The run index is explicitly set to None! A random seed will be selected.')
+            run_index = tuple(np.random.choice((0, 1, 2), size=1))
```
Comment (KEggensperger) [inline hpobench/benchmarks/nas/nasbench_101.py]: You should use self.rng for this

Comment (KEggensperger) [reply hpobench/benchmarks/nas/nasbench_101.py]: And l151 should have happened before this

File: hpobench/benchmarks/nas/nasbench_101.py [hunk 2]
```diff
@@ -109,17 +132,52 @@ def objective_function(self, configuration: Union[CS.Configuration, Dict],
...
+        else:
+            raise ValueError(f'run index must be one of Tuple or Int, but was {type(run_index)}')
+
         self.benchmark.reset_tracker()
 
         self.rng = rng_helper.get_rng(rng, self_rng=self.rng)
```
Comment (KEggensperger) [inline hpobench/benchmarks/nas/nasbench_101.py]: move up

**Review by KEggensperger — COMMENTED** [review]
Looks good to me. Can be merged after fixing the codestyle error and comments.

File: hpobench/benchmarks/nas/nasbench_101.py (hunk 1 above)
Comment (PhMueller) [reply hpobench/benchmarks/nas/nasbench_101.py]: I've also changed this for the other nas benchmarks. (rng as first operation + None selects a seed randomly)

**Review by KEggensperger — APPROVED** [review]
Looks good. Can be merged once the tests pass.

**Pull request merged**